agents/temporal_difference.c

```diff
@@ -0,0 +1,126 @@
+#include <float.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "game.h"
+#include "temporal_difference.h"
+#include "util.h"
+
+void init_td_agent(td_agent_t *agent, unsigned int state_num, char player)
+{
+    agent->player = player;
+    agent->state_value = malloc(sizeof(float) * state_num);
+    if (!(agent->state_value))
+        exit(1);
+    for (unsigned int i = 0; i < state_num; i++)
+        agent->state_value[i] = get_score(hash_to_table(i), player);
+}
+
+void load_model(td_agent_t *agent,
+                unsigned int state_num,
+                const char *model_path)
+{
+    FILE *fptr = fopen(model_path, "rb");
+    if (!fptr) {
+        printf("could not open state value table, train first\n");
```

Comment:
The error messages should start with "Failed to." That is, "Failed to open state value table file."